store.go

```diff
@@ -105,7 +106,7 @@ func (s *Store) getOrCreateTypedKey(k Key, v Value, kt KeyType) *BRecord {
 					}
 				}
 			}
-			br = thing.(*BRecord)
+			br = (*BRecord)(thing)
```

Comment:
What version of go are you using?  I'm building from the tip of devel, and compiling fails with:

./store.go:109: cannot convert thing (type gotomic.Thing) to type *BRecord: need type assertion